elements/lisk-p2p/test/unit/peer_pool.ts

```diff
@@ -71,14 +81,539 @@ describe('peerPool', () => {
+			const applyNodeInfoOnPeerCalls = applyNodeInfoOnPeerStub.getCalls()
+				.length;
+
+			expect(applyNodeInfoOnPeerCalls).eql(peerPool.getPeers().length);
+		});
+	});
+
+	describe('#request', () => {
+		it('should call getUniqueOutboundConnectedPeers', async () => {
+			sandbox.stub(peerPool, 'requestFromPeer').resolves();
+			const getUniqueOutboundConnectedPeersStub = sandbox
+				.stub(peerPool, 'getUniqueOutboundConnectedPeers')
+				.returns([peerInfo]);
+			await peerPool.request(requestPacket);
+
+			expect(getUniqueOutboundConnectedPeersStub).to.be.calledOnce;
+		});
+
+		it('should call _peerSelectForRequest', async () => {
+			sandbox.stub(peerPool, 'requestFromPeer').resolves();
+			const peers = peerPool.getUniqueOutboundConnectedPeers();
+			const _peerSelectForRequestStub = sandbox
+				.stub(peerPool as any, '_peerSelectForRequest')
+				.returns([peerInfo]);
+
+			await peerPool.request(requestPacket);
+
+			expect(_peerSelectForRequestStub).to.be.calledWith({
+				peers,
+				nodeInfo: peerPool.nodeInfo,
+				peerLimit: 1,
+				requestPacket,
+			});
+		});
+
+		it('should throw error if no peers selected', async () => {
+			sandbox.stub(peerPool as any, '_peerSelectForRequest').returns([]);
+			let err;
+			try {
+				await peerPool.request(requestPacket);
+			} catch (error) {
+				err = error;
+			}
+
+			expect(err).to.be.instanceOf(RequestFailError);
+		});
+
+		it('should call requestFromPeer', async () => {
+			const requestFromPeerStub = sandbox
+				.stub(peerPool, 'requestFromPeer')
+				.resolves();
+			sandbox
+				.stub(peerPool as any, '_peerSelectForRequest')
+				.returns([peerInfo]);
+			await peerPool.request(requestPacket);
+
+			expect(requestFromPeerStub).to.be.calledOnce;
+		});
+	});
+
+	describe('#send', () => {
+		let _peerSelectForSendStub: any;
+		let sendToPeer: any;
+
+		beforeEach(async () => {
+			_peerSelectForSendStub = sandbox
+				.stub(peerPool as any, '_peerSelectForSend')
+				.returns([peerInfo]);
+			sendToPeer = sandbox.stub(peerPool, 'sendToPeer').resolves();
+		});
+
+		it('should call _peerSelectForSend', async () => {
+			await peerPool.send(messagePacket);
+
+			expect(_peerSelectForSendStub).to.be.calledOnce;
+		});
+
+		it('should call sendToPeer for each selected peer', async () => {
+			await peerPool.send(messagePacket);
+
+			expect(sendToPeer).to.be.calledOnceWithExactly(
+				messagePacket,
+				constructPeerIdFromPeerInfo(peerInfo),
+			);
+		});
+
+		it(`should emit event if sendToPeer fails`, async () => {
+			sendToPeer.throws();
+			try {
+				await peerPool.send(1 as any);
+			} catch (error) {}
+
+			expect(peerPool.emit).to.be.calledOnce;
+		});
+	});
+
+	describe('#requestFromPeer', () => {
+		it('should throw error if no peers in peerPool', async () => {
+			(peerPool as any)._peerMap = new Map();
+			let err;
+			try {
+				await peerPool.requestFromPeer(requestPacket, peerId);
+			} catch (error) {
+				err = error;
+			}
+
+			expect(err).to.exist;
```

Comment:
Same as above.

Reply:
`try catch` required as toBeRejectedWith and throws does not work.